### ISSUE TITLE
Experimental Design : Training Script for GAT

### ISSUE THREAD

**Issue by BrandonN24** [BrandonN24/GCN-GAT-ComparativeAnalysis#10]
Create a training script for GAT using the following parameters:
Training: - Adam optimizer - 
LR: 0.01, weight decay: 5e-4 
200 epochs, early stopping 
Use NLL Loss
Dropout: 0.5 
Hyperparameters:
 - Hidden dims: 64 (try different values but document which one you settle with)
 - Layers: 2, 3 
 - GAT heads: 4 (try 8 if needed)
 
Ensure that the following metrics are recorded
- Accuracy
- Training time
- Parameter Count
- Convergence
